components/admin_console/request_button/request_button.tsx

```diff
@@ -16,122 +14,130 @@ import WarningIcon from 'components/widgets/icons/fa_warning_icon';
  * its outcome as either success, or failure accompanied by the
  * `message` property of the `err` object.
  */
-export default class RequestButton extends React.PureComponent {
-    static propTypes = {
+type Props = {
+
+    /**
+     * ID to assign to the form
+     */
+    id?: string;
+
+    /**
+     * The action to be called to carry out the request.
+     */
+    requestAction: (
+        success: (data?: any) => void,
+        error: (error: any) => void
+    ) => void;
+
+    /**
+     * A component that displays help text for the request button.
+     *
+     * Typically, this will be a <FormattedMessage/>.
+     */
+    helpText?: React.ReactNode;
+
+    /**
+     * A component to be displayed on the button.
+     *
+     * Typically, this will be a <FormattedMessage/>
+     */
+    loadingText?: React.ReactNode;
+
+    /**
+     * A component to be displayed on the button.
+     *
+     * Typically, this will be a <FormattedMessage/>
+     */
+    buttonText: React.ReactNode;
+
+    /**
+     * The element to display as the field label.
+     *
+     * Typically, this will be a <FormattedMessage/>
+     */
+    label?: React.ReactNode;
+
+    /**
+     * True if the button form control should be disabled, otherwise false.
+     */
+    disabled?: boolean;
+
+    /**
+     * True if the config needs to be saved before running the request, otherwise false.
+     *
+     * If set to true, the action provided in the `saveConfigAction` property will be
+     * called before the action provided in the `requestAction` property, with the later
+     * only being called if the former is successful.
+     */
+    saveNeeded?: boolean;
+
+    /**
+     * Action to be called to save the config, if saveNeeded is set to true.
+     */
+    saveConfigAction?: (callback: () => void) => void;
+
+    /**
+     * True if the success message should be show when the request completes successfully,
+     * otherwise false.
+     */
+    showSuccessMessage?: boolean;
+
+    /**
+     * The message to show when the request completes successfully.
+     */
+    successMessage: {
```

Comment:
this one was optional, but I guess since we provide this `successMessage` in default props there is no problem making it required so we don't have to check if `successMessage` is defined to access the inner properties.